File: 1/2021.06.30/151.Reverse Words in a String/java/solution.java
class Solution {
    public String reverseWords(String s) {
        List<String> list = new ArrayList<>();
        int i = 0;
        while (i < s.length()) {
            if (s.charAt(i) == ' ') {
                i++;
            } else {
                int j = i + 1;
                while (j < s.length()) {
                    if (s.charAt(j) == ' ') {
                        break;
                    } else {
                        j++;
                    }
                }
                list.add(s.substring(i, j));
                i = j;
            }
        }
        StringBuilder sb = new StringBuilder();
        for (i = list.size() - 1; i >= 0; i--) {
            sb.append(list.get(i));
            if (i != 0) {
                sb.append(" ");
            }
        }
        return sb.toString();
    }
}

File: 1/2021.06.30/151.Reverse Words in a String/python/solution.py
class Solution:
    def reverseWords(self, s: str) -> str:
        res = []
        i = 0
        while i < len(s):
            if s[i] == ' ':
                i += 1
            else:
                j = i + 1
                while j < len(s):
                    if s[j] == ' ':
                        break
                    else:
                        j += 1
                res.append(s[i:j])
                i = j
        return ' '.join(res[::-1])
